exporter/signalfxexporter/exporter_test.go

```diff
@@ -1048,3 +1050,220 @@ func TestSignalFxExporterConsumeMetadata(t *testing.T) {
+	}{
+		{
+			name: "Ingest CA not set",
+			config: &Config{
+				ExporterSettings: config.NewExporterSettings(component.NewID(typeStr)),
+				APIURL:           serverURL,
+				IngestURL:        serverURL,
+				AccessToken:      "random",
+				SyncHostMetadata: true,
+			},
+			wantErr:        true,
+			wantErrMessage: "certificate is not trusted",
+		},
+		{
+			name: "Ingest CA set",
+			config: &Config{
+				ExporterSettings: config.NewExporterSettings(component.NewID(typeStr)),
+				APIURL:           serverURL,
+				IngestURL:        serverURL,
+				IngestCAFile:     "./testdata/certs/ca.pem",
+				AccessToken:      "random",
+				SyncHostMetadata: true,
+			},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			sfx, err := newSignalFxExporter(tt.config, zap.NewNop())
+			assert.NoError(t, err)
+
+			_, err = sfx.pushMetricsData(context.Background(), metricsPayload)
+			if tt.wantErr {
+				require.Error(t, err)
+				if tt.wantErrMessage != "" {
+					require.ErrorContains(t, err, tt.wantErrMessage)
+				}
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestTLSAPIConnection(t *testing.T) {
+	observedZapCore, observedLogs := observer.New(zap.DebugLevel)
+	logger := zap.New(observedZapCore)
+	metadata := []*metadata.MetadataUpdate{
+		{
+			ResourceIDKey: "key",
+			ResourceID:    "id",
+			MetadataDelta: metadata.MetadataDelta{
+				MetadataToAdd: map[string]string{
+					"prop.erty1": "val1",
+				},
+			},
+		},
+	}
+
+	server, err := NewLocalHTTPSTestServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, "connection is successful")
+	}))
+	require.NoError(t, err)
+	defer server.Close()
+
+	tests := []struct {
+		name           string
+		config         *Config
+		wantErr        bool
+		wantErrMessage string
+	}{
+		{
+			name: "API CA set",
+			config: &Config{
+				ExporterSettings: config.NewExporterSettings(component.NewID(typeStr)),
+				APIURL:           "tmp",
+				IngestURL:        "tmp",
+				AccessToken:      "random",
+				SyncHostMetadata: true,
+				APICAFile:        "./testdata/certs/ca.pem",
+			},
+			wantErr: false,
+		},
+		{
+			name: "API CA set",
+			config: &Config{
+				ExporterSettings: config.NewExporterSettings(component.NewID(typeStr)),
+				APIURL:           "tmp",
+				IngestURL:        "tmp",
+				AccessToken:      "random",
+				SyncHostMetadata: true,
+			},
+			wantErr:        true,
+			wantErrMessage: "certificate is not trusted",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.config.APIURL = server.URL
+			sfx, err := newSignalFxExporter(tt.config, logger)
+			assert.NoError(t, err)
+			err = sfx.pushMetadata(metadata)
+			// We can't control the default send delay of 10 seconds
+			// I am afraid that this is adding significant delay to the test suite
+			// I can make the send delay configurable, or take out this test.
+			time.Sleep(12 * time.Second)
```

Comment:
Is this because the host metadata are being send with 10 seconds delay? Can mutate the variable in the test runtime instead of exposing it as a config option

Reply:
I am trying to test the sfx object initialization with the user exposed config and the delay is not exposed, unless I shortcut this and create a dimension client with the TLS config, but that misses part of the test I am trying to target

Reply:
with configtls, it was cleaner to use `NewDimensionClient` and be able to set the send delay.